tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py

```diff
@@ -125,26 +82,12 @@ def tap_collection(self, collection_name):
 
     @property
     def app_elements(self):
-        return self.marionette.execute_script("""
-        var gridItems = window.wrappedJSObject.app.grid.getItems();
-        var appElements = [];
-        for(var i=0; i<gridItems.length; i++){
-        // it must have an app to be a
-        if(gridItems[i].app) appElements.push(gridItems[i].element);
-        }
-        return appElements;
-        """)
-
-    @property
-    def divider_elements(self):
-        return self.marionette.find_elements(*self._divider_locator)
+        return [self.HomescreenElement(self.marionette, app_element)
+                for app_element in self.marionette.find_elements(*self._all_icons_locator)]
```

Comment:
HomescreenElement deals with the needed switch to shadow root. HomescreenElement is a representation of 1 web component